controllers/imageclusterinstall_controller.go

```diff
@@ -753,35 +773,51 @@ func (r *ImageClusterInstallReconciler) writeInputData(
 			return err
 		}
 		clusterInfo := r.getClusterInfoFromFile(clusterInfoFilePath)
+		if clusterInfo == nil {
+			clusterInfo = &lca_api.SeedReconfiguration{}
+		}
 
 		crypto, err := r.Credentials.EnsureKubeconfigSecret(ctx, cd, clusterInfo)
 		if err != nil {
 			return fmt.Errorf("failed to ensure kubeconifg secret: %w", err)
 		}
 
-		kubeadminPasswordHash, err := r.Credentials.EnsureAdminPasswordSecret(ctx, cd)
+		kubeadminPasswordHash, err := r.Credentials.EnsureAdminPasswordSecret(ctx, cd, clusterInfo.KubeadminPasswordHash)
 		if err != nil {
 			return fmt.Errorf("failed to ensure admin password secret: %w", err)
 		}
-		if err := r.writeClusterInfo(ctx, log, ici, cd, crypto, psData, kubeadminPasswordHash, clusterInfoFilePath, clusterInfo, bmh); err != nil {
+
+		newClusterInfo, err := r.createSeedReconfiguration(ctx, log, ici, cd, crypto, psData, kubeadminPasswordHash, clusterInfo, bmh)
+		if err != nil {
+			return fmt.Errorf("failed to create seed reconfiguration: %w", err)
+		}
+
+		if err := r.writeClusterInfo(newClusterInfo, clusterInfoFilePath); err != nil {
```

Comment:
We probably now don't need to split these, right?

And if you want to optimize out the file write if nothing changed then you can do the deep equal check in the function before writing it out, right?

Reply:
yep as it was before

Reply:
I mean you can recombine `createSeedReconfiguration` and `writeClusterInfo` into a single function. Is there any reason for them to be separate?

Reply:
ooo, right , when i removed the logic i forgot that i split it